### ISSUE TITLE
Use AWS Spot Describe Spot Pricing history for more accurate estimates until the spot data feed is available

### ISSUE THREAD

**Issue by AjayTripathy** [kubecost/cost-model#784]
Today, until data for AWS spot price nodes becomes available in the spot data feed:
1. kubecost uses the public pricing API estimate for a node if you have not supplied a tag for the spot node
2. a flat per CPU/ GB RAM estimate (tunable (here)[https://github.com/kubecost/cost-analyzer-helm-chart/blob/v1.79.1/cost-analyzer/values.yaml#L589] ) if a spot node has been tagged and kubecost has been made aware of the (tags)[https://github.com/kubecost/cost-analyzer-helm-chart/blob/v1.79.1/cost-analyzer/values.yaml#L605]

Meaning our estimates can be off for several hours (the speed at which AWS updates this feed). The estimate continues to be off until we reconcile prices with the AWS CUR. 

Instead, we could use instancetype + region to get much more accurate estimates until CUR data becomes available by integrating with this API:

https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeSpotPriceHistory.html

**Comment by michaelmdresser** [comment]
@AjayTripathy do you know what the status of this is? Have we made progress? Just trying to keep everything up-to-date :smile:

**Comment by AjayTripathy** [comment]
No progress but still something we want to do...

**Comment by dudicoco** [comment]
@AjayTripathy wouldn't using the `DescribeSpotPriceHistory` API also remove the need to [export spot data feed to S3](https://docs.kubecost.com/install-and-configure/install/cloud-integration/aws-cloud-integrations/aws-spot-instances)?

**Comment by AjayTripathy** [comment]
Yes it would...however this may be out of scope since the plan of opencost is not to support cloud-provider specific pricing long term-- it's a better fit for the community edition Kubecost.
